porch/pkg/engine/engine.go

```diff
@@ -374,6 +383,11 @@ func ensureSameOrigin(ctx context.Context, repo repository.Repository, obj *api.
 		// If there are no tasks, or the first task is not init or clone, then this revision was not
 		// created from another package revision. That means we expect it to be the first revision
 		// for this package.
+		if len(tasks) == 0 {
+			klog.Warningf("not same origin due to no tasks")
```

Comment:
Is it intentional that we add this logging in this PR?

Reply:
I can remove it. This PR does not fix ensureSameOrigin; I think I will disable in a separate PR.